src/widgets/chart.rs

```diff
@@ -904,17 +908,13 @@ impl<'a> Chart<'a> {
         Rect::new(min_x, y, max_x - min_x, 1)
     }
 
-    fn render_label(buf: &mut Buffer, label: &Span, label_area: Rect, alignment: Alignment) {
-        let label_width = label.width() as u16;
-        let bounded_label_width = label_area.width.min(label_width);
-
-        let x = match alignment {
-            Alignment::Left => label_area.left(),
-            Alignment::Center => label_area.left() + label_area.width / 2 - bounded_label_width / 2,
-            Alignment::Right => label_area.right() - bounded_label_width,
+    fn render_label(buf: &mut Buffer, label: &Line, label_area: Rect, alignment: Alignment) {
+        let label = match alignment {
+            Alignment::Left => label.clone().left_aligned(),
+            Alignment::Center => label.clone().centered(),
+            Alignment::Right => label.clone().right_aligned(),
```

Comment:
There are usually a fairly small number of short labels here, so the clone is probably ok here.

Reply:
Why is it needed in the first place? :thinking: do the alignment methods require ownership?

Reply:
Ah, it moves the original… well…